express/schemas/manifest.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "metadata": {
+      "type": "object",
+      "properties": {
+        "base_path": {
+          "type": "string",
+          "format": "uri"
+        }
+      },
+      "required": [
+        "base_path"
+      ]
+    },
+    "index": {
+      "type": "object",
+      "properties": {
+        "location": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "location"
+      ]
+    },
+    "subsets": {
+      "$ref": "#/definitions/subsets"
+    }
+  },
+  "required": [
+    "metadata",
+    "index",
+    "subsets"
+  ],
+  "definitions": {
+    "field": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": ["str", "int", "float", "bool", "bytes"]
```

Comment:
Currently used the most relevant Python types here. Do we need additional types such as `vector`?  Any existing type systems we can leverage?

Reply:
Vectors are typically stored as `float`, so I guess we can use that one

Reply:
The vector would be an array of `float`s though, so the type of the column is not `float`.

Reply:
Ok yes `list` should be supported, as well as `tuple` and `dict`

Reply:
Data in `parquet` is not stored as a `list`, `tuple`, or `dict`.

We can probably leverage the parquet types:
- https://parquet.apache.org/docs/file-format/types/
- https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
- https://learn.microsoft.com/en-us/common-data-model/sdk/parquet-to-cdm-datatype

Reply:
Indeed the types are probably not one-to-one mapping to the python data types but have their own notations (probably byte array). 

https://arrow.apache.org/docs/python/generated/pyarrow.ListArray.html

Reply:
For component validating:

do we validate the parquet schema or load it into something python and validate?

Reply:
What I currently had in my mind is:
- We validate the manifests and components against each others. The used type system doesn't really matter here as long as it's consistent across all manifests and components.
- We instruct the dataframe library to load the data from parquet. I think we can provide a schema if we want the library to validate the parquet schema, otherwise it will just infer it.

Not sure at which level we need to define the types. Referring to the [parquet docs](https://parquet.apache.org/docs/file-format/types/), I think we need to define it on the logical level, since that will be important for the components, not the underlying storage type.
- PyArrow specifies logical types ([docs](https://arrow.apache.org/docs/python/api/datatypes.html#api-types)).
- So does eg. Polars ([docs](https://pola-rs.github.io/polars/py-polars/html/reference/datatypes.html))

They mostly match and I based the current list on these.

Following a dataframe library would be easiest, but probably also limits us to that framework which we might not want to do on the long term. `PyArrow` is used by most of these frameworks, so that seems like a good basis for multi-library compatibility.